internal/grid/connection.go

```diff
@@ -1578,11 +1601,30 @@ func (c *Connection) State() State {
 }
 
 // Stats returns the current connection stats.
-func (c *Connection) Stats() ConnectionStats {
-	return ConnectionStats{
-		IncomingStreams: c.inStream.Size(),
-		OutgoingStreams: c.outgoing.Size(),
-	}
+func (c *Connection) Stats() madmin.RPCMetrics {
+	conn := 0
+	if c.State() == StateConnected {
+		conn++
+	} else {
+		conn--
+	}
+	m := madmin.RPCMetrics{
+		CollectedAt:      time.Now(),
+		Connected:        conn,
```

Comment:
can this field have a negative value ? I am not sure if I understand the logic of conn

Reply:
Yeah. Looks like a brainfart. Fixing